src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -38,6 +38,8 @@
     private long committedEvents;
     private long sentEvents;
     private long batchesSent;
+    // Uncommitted offsets are calculated right on exiting from Streaming state.
+    private Map<Partition.PartitionKey, Long> uncommittedOffsets;
```

Comment:
Maybe you could initialise it with an empty `Map` so that there will be no need for null checks.